src/include/miopen/graphapi/graphapi_tensor.hpp

```diff
@@ -0,0 +1,190 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#pragma once
+
+#include <miopen/graphapi/graphapi.hpp>
+
+#include <cstdint>
+#include <vector>
+
+namespace miopen {
+
+namespace graphapi {
+
+class Tensor
+{
+public:
+    Tensor() noexcept         = default;
+    Tensor(const Tensor&)     = default;
+    Tensor(Tensor&&) noexcept = default;
+    Tensor& operator=(const Tensor&) = default;
+    Tensor& operator=(Tensor&&) noexcept = default;
+    Tensor(miopenDataType_t dataType,
+           const std::vector<int64_t>& dimensions,
+           const std::vector<int64_t>& strides,
+           int64_t id,
+           bool isVirtual)
+        : mDimensions(dimensions),
+          mStrides(strides),
+          mId(id),
+          mDataType(dataType),
+          mVirtual(isVirtual)
+    {
+    }
+    Tensor(miopenDataType_t dataType,
+           std::vector<int64_t>&& dimensions,
+           std::vector<int64_t>&& strides,
+           int64_t id,
+           bool isVirtual) noexcept
+        : mDimensions(std::move(dimensions)),
+          mStrides(std::move(strides)),
+          mId(id),
+          mDataType(dataType),
+          mVirtual(isVirtual)
+    {
+    }
+
+    miopenDataType_t getDataType() const noexcept { return mDataType; }
+    const std::vector<int64_t>& getDimensions() const noexcept { return mDimensions; }
+    const std::vector<int64_t>& getStrides() const noexcept { return mStrides; }
+    int64_t getId() const noexcept { return mId; }
+    bool getVirtual() const noexcept { return mVirtual; }
+
+    Tensor& setDataType(miopenDataType_t dataType) noexcept
```

Comment:
we don't need `setters` here because that's what the builder is for.

Reply:
I saw a TensorParameters in cudnn_frontend class with these capabilities. I think we will be able to reuse these setters.

Reply:
Usually setters open the door to creating a logical invalid state in an object so I recommend we don't open that door and everyone goes through the builder route.

Reply:
OK makes sense

Reply:
Done